Text/Megaparsec/Byte/MachineInts.hs

```diff
@@ -0,0 +1,110 @@
+-- TODO debug
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- | "Machine" integer parsers.
+module Text.Megaparsec.Byte.MachineInts where
+
+import Data.Bits
+-- TODO debug below
+import qualified Data.ByteString as BS
+import Data.Maybe (fromMaybe)
+import Data.Proxy (Proxy (..))
+import Data.Void (Void)
+import Data.Word
+import Text.Megaparsec
+import Text.Megaparsec.Byte.MachineInts.Internal
```

Comment:
I think it is better to merge the two modules.

Reply:
Yep, this was primarily to separate Megaparsec functions from utility ones. No need if it's actually in Megaparsec.